perception/sub8_perception/nodes/test_path_marker.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+PKG = 'test_path_marker'
+import rosbag
+import rospy
+import sys
+import unittest
+import numpy as np
+from sub8_msgs.srv import VisionRequest2D, VisionRequest2DRequest, VisionRequest2DResponse
+from sensor_msgs.msg import Image, CameraInfo
+
+class TestPathMarker(unittest.TestCase):
+    def __init__(self, *args):
+        rospy.init_node("path_marker_test_py", anonymous=True)
+        rospy.wait_for_service('/vision/path_marker/2D', 3.0)
+        self.service = rospy.ServiceProxy('/vision/path_marker/2D', VisionRequest2D) 
+        self.cam_info_pub = rospy.Publisher('/down_camera/camera_info', CameraInfo, queue_size=5)
+        self.img_pub = rospy.Publisher('/down_camera/image_rect_color', Image, queue_size=5)
+        super(TestPathMarker, self).__init__(*args)
+
+    def get_ideal_response(self, pts):
+        pts = np.array(pts)
+        res = VisionRequest2DResponse()
+        middle = pts[0]+(pts[1]-pts[0])/2
+        theta = np.arctan( (float(pts[1][1])-pts[0][1]) / (pts[1][0] - pts[0][0]) )
+        res.pose.x = middle[0]
+        res.pose.y = middle[1]
+        res.pose.theta = theta
+        return res
+
+    def _test_bag(self, filename, duration, pts):
+        rospy.sleep(rospy.Duration(0.5))
+        correct = self.get_ideal_response(pts)
+        print "Correct pose: {}".format(correct.pose.theta/(2*np.pi)*360.0)
+        bag = rosbag.Bag(filename)
+        first = True
+        first_time = None
+        previous_time = None
+        i = 0
+        for topic, msg, t in bag.read_messages():
+            if first:
+                first_time = t
+                previous_time = t
+                first = False
+            else:
+                rospy.sleep(t-previous_time) # Slow down to watch happen
+                previous_time = t
+                if t - first_time > duration:
+                    did = True
+                    break
+            if topic == "/down_camera/image_rect_color":
+                self.img_pub.publish(msg)
+            if topic == "/down_camera/camera_info":
+                self.cam_info_pub.publish(msg)
+        res = self.service(VisionRequest2DRequest())
+        # 10 pixel error accepted
+        res_xy = np.array([res.pose.x, res.pose.y])
+        correct_xy = np.array([correct.pose.x, correct.pose.y])
+        err = np.linalg.norm(res_xy-correct_xy)
+        msg="Marker pose (x,y) too much error Res={} Correct={} Error={}".format(res_xy, correct_xy, err)
+        self.assertLess(err, 
+                        10.0,
+                        msg=msg)
+        #5 degrees error
+        theta_err = abs(res.pose.theta-correct.pose.theta)
```

Comment:
You can't subtract angles like this in the general case. Make sure the possible angles here are bounded in [-pi/2, pi/2] for this operation to be valid.